ocp.tf

```diff
@@ -8,13 +8,14 @@ provider "ibm" {
 
 resource "random_id" "label" {
   count       = var.cluster_id == "" ? 1 : 0
-  byte_length = "2" # Since we use the hex, the word lenght would double
+  byte_length = "2" # Since we use the hex, the word length would double
   prefix      = "${var.cluster_id_prefix}-"
 }
 
 locals {
   # Generates cluster_id as combination of cluster_id_prefix + (random_id or user-defined cluster_id)
   cluster_id   = var.cluster_id == "" ? random_id.label[0].hex : (var.cluster_id_prefix == "" ? var.cluster_id : "${var.cluster_id_prefix}-${var.cluster_id}")
+  name_prefix  = var.use_zone_info_for_instance ? "${local.cluster_id}-${var.ibmcloud_zone}" : local.cluster_id
```

Comment:
1. Rename it to `name_suffix`
2. Just pass the suffix part and use `"${var.cluster_id}${var.name_suffix}"` everywhere.
```
name_suffix = var.use_zone_info_for_instance ? "-${var.ibmcloud_zone}" : ""
```

This will ensure the existing flow for cluster_id is not impacted. For example no_proxy, bastion name for external DNS,. etc.

Reply:
I am even okay with `cluster_id_with_zone` if you are planning to include `local.cluster_id` to it. This will also help in validating the length of 32 with zone info.